app/assets/javascripts/basetypes/DataElement.js

```diff
@@ -13,15 +14,20 @@ function DataElementSchema(add, options) {
     extended.add(add);
   }
 
-  // Returns all of the codes on this data element
-  // in a format usable by the cql-execution framework
+  // Returns all of the codes on this data element in a format usable by
+  // the cql-execution framework.
```

Comment:
does there need to be more differentiation with the naming of these functions?

`getCode` returns *a list* of codes usable by the execution framework 
`code` returns *a single code* usable by the execution framework
`codes` returns a list of codes not usable by the execution framework (at least w no transforms).

The naming of these are not explanatory and not consistent. `getCode` singular for returning a list. `get...` returns a transform and `code` returns a transform, but then `codes` returns no transform.

Reply:
`codes` should **definitely** be removed, as it is useless.

`codes` is useful for us only, as it is a way to get all of the codes from a data element in our own format.

`getCode` is annoyingly named, but is necessary, because that is the function (verbatim) that is called by the execution engine.

Reply:
These could **definitely** use better comment explanations though!

Reply:
Apparently `code` and `getCode` are both used by the execution engine. So at least on the front end, they should both be retained.